src/Core/Services/TypeHelper.cs

```diff
@@ -43,6 +44,7 @@ public static class TypeHelper
             [typeof(bool?)] = DbType.Boolean,
             [typeof(char?)] = DbType.StringFixedLength,
             [typeof(Guid?)] = DbType.Guid,
+            [typeof(TimeSpan?)] = DbType.Time,
```

Comment:
My PR has a helper you can mirror so that adding nullable types to this dictionary isn't necessary: https://github.com/Azure/data-api-builder/blob/04319b4da04fd1a1590ccc67f345b17522aa7177/src/Core/Services/TypeHelper.cs#L173-L203